src/S3/MultipartCopy.php

```diff
@@ -178,20 +187,45 @@ private function fetchSourceMetadata()
         if ($this->config['source_metadata'] instanceof ResultInterface) {
             return $this->config['source_metadata'];
         }
-
-        list($bucket, $key) = explode('/', ltrim($this->source, '/'), 2);
-        $headParams = [
-            'Bucket' => $bucket,
-            'Key' => $key,
-        ];
-        if (strpos($key, '?')) {
-            list($key, $query) = explode('?', $key, 2);
-            $headParams['Key'] = $key;
-            $query = Psr7\parse_query($query, false);
-            if (isset($query['versionId'])) {
-                $headParams['VersionId'] = $query['versionId'];
+        if (is_array($this->source)) {
+            $headParams['Key'] = $this->source['source_key'];
```

Comment:
`$headParams` is undefined at the point when this key is set, which I think generates warnings in some versions.

```php
$headParams = [
    'Bucket' => $this->source['source_bucket'],
    'Key' => $this->source['source_key'],
];
```

Reply:
nit: Any reason this is defined over 3 statements still?

Reply:
Good call.  Now it's only one statement :)